dds_web/scheduled_tasks.py

```diff
@@ -1,24 +1,74 @@
 import flask_apscheduler
 import flask
 
-# Apscheduler
+## Apscheduler
 scheduler = flask_apscheduler.APScheduler()
 
 
+# @scheduler.task("cron", id="available_to_expired", minute=5, hour=2, misfire_grace_time=1)
 @scheduler.task("interval", id="available_to_expired", seconds=10, misfire_grace_time=1)
-def change_status_to_expired():
-    # print("Task: Change project status from Available to expired.", flush=True)
-    from dds_web.database import models
+def set_available_to_expired():
+    scheduler.app.logger.debug("Task: Checking for Expiring projects.")
+    import sqlalchemy
 
-    # from dds_web.utils import current_time, page_query
+    from dds_web import db
+    from dds_web.database import models
+    from dds_web.errors import DatabaseError
+    from dds_web.api.project import ProjectStatus
+    from dds_web.utils import current_time, page_query
 
     with scheduler.app.app_context():
-        # scheduler.app.logger.debug("This means that the app context works!")
-        test = models.Project.query.all()
-        # print(test, flush=True)
+        expire = ProjectStatus()
+
+        errors = {}
+
+        for unit in models.Unit.query.all():
```

Comment:
We also need to add a try-except around this for loop so that we can log the error
```
try:
    for unit.... etc and do stuff
except (both operational and sql):
    log the error 
    rollback 
    "raise"
```

Reply:
We can improve on it later maybe and have it as a decorator or something but just add this now so that we can catch potential query-issues